File: backend/application/models/room.py
from application import db
from application.models.maintenance import *
from application.models.note import *
from application.models.booking import *

class Room(db.Document):
	name = db.StringField()
	created_at = db.DateTimeField()
	modified_at = db.DateTimeField()
	building = db.StringField()
	is_available = db.BooleanField()
	room_number = db.StringField()
	current_booking = db.ReferenceField(Booking,reverse_delete_rule=mongoengine.NULLIFY)
	last_active = db.DateTimeField()
	maintenance_history = db.ListField(db.ReferenceField(Maintenance,reverse_delete_rule=mongoengine.PULL))
	booking_history = db.ListField(db.ReferenceField(Booking,reverse_delete_rule = db.PULL))

File: frontend/src/app/models/Room.ts
import {Booking} from './Booking'
export class Room {
    room_number: string;
    name: string;
    building :string;
    is_available : any;
    room_state: string;
}

export class RoomDetail {
	'booking_history': Array<Booking>;
  'building' : string;
  'current_booking': Booking;
  'description': string;
  'maintenance_history':Array<any>;
  'modified_at': string;
  "name": string;
  'room_number': string;
  'room_state' : number;
}
